Gems/Multiplayer/Code/Source/Components/LocalPredictionPlayerInputComponent.cpp

```diff
@@ -312,30 +352,39 @@ namespace Multiplayer
         SerializeEntityCorrection(serializer);
         GetNetBindComponent()->NotifyCorrection();
 
-#ifndef AZ_RELEASE_BUILD
+        const uint32_t inputHistorySize = static_cast<uint32_t>(m_inputHistory.Size());
+        const uint32_t historicalDelta = aznumeric_cast<uint32_t>(m_clientInputId - inputId); // Do not replay the move we just corrected, that was already processed by the server
```

Comment:
nit: avoid personal  pronouns in comments 'we' is often unclear to the reader. Also as this code can be run in game, server or editor again, the we is unclear.